spatialos-sdk/src/worker/component.rs

```diff
@@ -139,23 +130,22 @@ impl<'a> ComponentDataRef<'a> {
         }
     }
 
-    // NOTE: We manually declare that the component impl `TypeConversion` and `Clone`
+    // NOTE: We manually declare that the component impl `ObjectField`
     // here, but in practice this will always be true for all component types. Future
     // iterations should clean this up such that the `Component` trait can imply these
     // other bounds automatically (i.e. by making them super traits of `Component`).
```

Comment:
Is it worth doing this note in this PR?

Reply:
Yeah, that would be a good idea 😁

Reply:
Updated `Component` to be a sub-trait of `ObjectField`. That removed the need for `from_data` and `to_data`. I also added some helper methods to `SchemaComponentData` to make serialization easier, and removed the trait bounds noted in this comment.